demos/healthcare_assistant_demo/main.py

```diff
@@ -71,57 +72,95 @@ def load_asr_model(model_name: str) -> None:
 
+        log.info(f"Quantizing {model_name} to INT4... It may take significant amount of time depending on your machine power.")
         quantizer = OVQuantizer.from_pretrained(chat_model, task="text-generation")
         quantizer.quantize(save_directory=model_path, weights_only=True, ov_config=config)
 
         chat_tokenizer = AutoTokenizer.from_pretrained(model_name, token=token)
         chat_tokenizer.save_pretrained(model_path)
 
-    return OpenVINOLLM(context_window=2048, model_id_or_path=str(model_path), max_new_tokens=512, device_map=device,
+    return OpenVINOLLM(context_window=2048, model_id_or_path=str(model_path), max_new_tokens=512, device_map="AUTO:GPU,CPU",
                        model_kwargs={"ov_config": ov_config}, generate_kwargs={"do_sample": True, "temperature": 0.7, "top_k": 50, "top_p": 0.95})
 
 
+def optimize_model_for_npu(model: OVModelForFeatureExtraction):
+    class ReplaceTensor(passes.MatcherPass):
+        def __init__(self, packed_layername_tensor_dict_list):
+            super().__init__()
+            self.model_changed = False
+
+            param = passes.WrapType("opset10.Multiply")
+
+            def callback(matcher: passes.Matcher) -> bool:
+                root = matcher.get_match_root()
+                if root is None:
+                    return False
+                for y in packed_layername_tensor_dict_list:
+                    root_name = root.get_friendly_name()
+                    if root_name.find(y["name"]) != -1:
+                        max_fp16 = np.array([[[[-np.finfo(np.float16).max]]]]).astype(np.float32)
+                        new_tenser = ops.constant(max_fp16, ov.Type.f32, name="Constant_4431")
+                        root.set_arguments([root.input_value(0).node, new_tenser])
+                        packed_layername_tensor_dict_list.remove(y)
+
+                return True
+
+            self.register_matcher(passes.Matcher(param, "ReplaceTensor"), callback)
+
+    packed_layer_tensor_dict_list = [{"name": "aten::mul/Multiply"}]
+
+    manager = passes.Manager()
+    manager.register_pass(ReplaceTensor(packed_layer_tensor_dict_list))
+    manager.run_passes(model.model)
+    model.reshape(1, 512)
+
+
 def load_embedding_model(model_name: str) -> OpenVINOEmbedding:
     model_path = MODEL_DIR / model_name
 
     if not model_path.exists():
-        embedding_model = OVModelForFeatureExtraction.from_pretrained(model_name, export=True, device="AUTO")
+        embedding_model = OVModelForFeatureExtraction.from_pretrained(model_name, export=True)
+        optimize_model_for_npu(embedding_model)
         embedding_model.save_pretrained(model_path)
         embedding_tokenizer = AutoTokenizer.from_pretrained(model_name)
         embedding_tokenizer.save_pretrained(model_path)
 
-    return OpenVINOEmbedding(str(model_path))
+    return OpenVINOEmbedding(str(model_path), device="AUTO:NPU,CPU", model_kwargs={"dynamic_shapes": False})
```

Comment:
Whats the purpose of `model_kwargs={"dynamic_shapes": False}`

Reply:
The code here takes it into acount:
https://github.com/huggingface/optimum-intel/blob/e9800ced0f6ceaa7aa0afe67327bfe348815620d/optimum/intel/openvino/modeling_base.py#L75